lib/galaxy/jobs/handler.py

```diff
@@ -395,6 +398,10 @@ def __monitor_step(self):
             self.__handle_waiting_jobs()
         except StopSignalException:
             pass
+        except DBAPIError as e:
```

Comment:
I don't think that catches https://sentry.galaxyproject.org/organizations/galaxy/issues/142454/?project=3&query=is%3Aunresolved&referrer=issue-stream&sort=freq&statsPeriod=14d, which is an `InvalidRequestError`. Maybe we use the widest exception which seems to be `SQLAlchemyError` ?

Reply:
Or we don't do this at all but remove the session in a finally statement ? That's less custom logic and seems like the most robust solution ?